DOC-13056: Antora assembler simplification

osfameron opened the pull request:
WIP with antora-assembler

Android database page looks pretty good to me, but haven't done full test!

osfameron committed:
First pass for simplification of complex partials

Run
    perl -ni -e 'print unless /^(\s*$|\/\/)/' \
         modules/ROOT/pages/_partials/_show_glossary_links.adoc
         ...

on the main abstracted partials. This is so that when antora-assembler inlines them,
they won't have extra newlines and // comment lines that confuse the output.

Additionally, manually add a comment to the block_tabbed_code partials, so that
the source of includes can be tracked.

osfameron committed:
simplify script

run:

  ./simplify/simplify.sh

This assumes that docs-site and docs-sync-gateway are siblings in the file tree.

This sample only updates the android/database.adoc page for now
(it builds everything, and all the rest should work in exactly the same way,
I just haven't wired that part together yet.)

osfameron committed:
Run improved simplify script on all modules

* Extract `process` call to run on all modules, and all asciidoc files

* {tabs} declarations get mangled
  (possibly same issue as https://gitlab.com/antora/antora-assembler/-/issues/17 ?)
  We now de-munge these

osfameron committed:
Result of running current simplify.sh

osfameron committed:
demunge plantuml

osfameron committed:
improve demunging

osfameron committed:
update modules

osfameron committed:
demunge additional tabs anchors

these auto-generated anchors break tabs

osfameron committed:
fix tabs

osfameron committed:
Tidy partials

Remove // comments where unneeded

osfameron committed:
More fixup (remove comments etc.)

osfameron committed:
Updated output

osfameron committed:
Tidy output

osfameron committed:
Update outputs

osfameron committed:
tidy

osfameron committed:
back out one change

osfameron committed:
update output

osfameron committed:
remove commented include

osfameron committed:
more updates

osfameron committed:
update output

RayOffiah reviewed (approved):
Haven't been through all the files, to be honest; but it looks like you've just removed all the `discrete` tags from the anchors.

So it should be good!

Ah! My mistake; got here a little too late.

osfameron committed:
Update simplification for 3.2.1

reran simplify.sh

osfameron committed:
update simplify scripts

better output and handling of edge cases

osfameron committed:
Merge branch 'release/3.2' into antora-assembler-simplification

osfameron committed:
fix tag I broke

osfameron committed:
fix trailing + in attribute

causes problems when interpolated in post-processing

osfameron committed:
remove passthrough mangling

osfameron committed:
tag snippets in ROOT

osfameron committed:
demunge tabs (better) and images

osfameron committed:
update outputs

simon-dew committed:
Add peer-to-peer to C nav

simon-dew committed:
Format tweaks for peer-to-peer common partial

osfameron committed:
update simplify

osfameron committed:
update sources

osfameron committed:
remove unused page

osfameron committed:
update code snippets partials

osfameron committed:
work on code source include

osfameron committed:
update source code processing

osfameron committed:
update sources

osfameron committed:
Remove overridden nav titles (for simplification)

As Antora Assembler uses the nav title as the header,
removing the overridden titles from nav will prevent
Assembler from changing the titles!

:galaxy-brain: thinking from @simon-dew

osfameron committed:
keep source-language

osfameron committed:
DOC-13056 iterate simplification

from @simon-dew comments

osfameron committed:
updating sources

osfameron committed:
Fix weird anchor mangling

osfameron committed:
update sources

osfameron committed:
PLEASE DO EDIT

(remove "DO NOT EDIT" from comments)

osfameron committed:
update simplification

osfameron committed:
update sources

osfameron committed:
update source examples for gs-*

osfameron committed:
demangle links

osfameron committed:
updated sources

osfameron committed:
remove maintenance:1

osfameron committed:
remove spurious reset of :page-toclevels:

osfameron committed:
Experimentally revert "optimization" to fix tab blocks

osfameron committed:
update sources

osfameron committed:
update sources

osfameron committed:
update sources

osfameron committed:
update image paths

osfameron committed:
update images

osfameron committed:
updated sources

osfameron committed:
Demangle pass:[] spaghetti

osfameron committed:
DOC-13226 remove random headings to API docs

The Conflict page is followed in nav by the API reference,
which are hardcoded links.

As antora-assembler handles those by just emitting them, they
don't get `split` in the expected way, so end up polluting
the preceding page.
This commit just filters them out.

osfameron committed:
DOC-13170 fix C urls

osfameron committed:
Update partials for 3.2.3

osfameron committed:
remove pages deleted by Simon

osfameron committed:
add new files for 3.2.2 and 3.2.3

osfameron committed:
more new files

osfameron committed:
comment out includes (to allow them to be uncommented later...)

osfameron committed:
updates fixing the indexing pages

osfameron committed:
3.2.3

osfameron committed:
update upgrade.adoc

osfameron committed:
fix query-live by removing query-troubleshooting

osfameron committed:
DOC-13262 C new logging API

osfameron committed:
C new-logging-api

osfameron committed:
new-logging-api
